pkg/provisioner/crdprovisioner.go

```diff
@@ -337,11 +336,20 @@ func (c *CrdProvisioner) PublishVolume(
 	readOnly bool,
 	secrets map[string]string,
 	volumeContext map[string]string) (map[string]string, error) {
-	azVA := c.azDiskClient.DiskV1alpha1().AzVolumeAttachments(c.namespace)
+
+	// check for node and volume
 	volumeName, err := azureutils.GetDiskNameFromAzureManagedDiskURI(volumeID)
 	if err != nil {
 		return nil, status.Errorf(codes.NotFound, "Error finding volume : %v", err)
 	}
+	if _, err := c.azDiskClient.DiskV1alpha1().AzDriverNodes(c.namespace).Get(ctx, strings.ToLower(nodeID), metav1.GetOptions{}); err != nil {
+		return nil, err
+	}
+	if _, err := c.azDiskClient.DiskV1alpha1().AzVolumes(c.namespace).Get(ctx, strings.ToLower(volumeName), metav1.GetOptions{}); err != nil {
+		return nil, err
+	}
+
```

Comment:
I am assuming that these do-nothing gets are simply to confirm existence of the target node and volume. We should trust that the CSI infra is not asking us to operate on non-existing nodes. As it is, these additional gets unnecessarily increase the QPS initiated by our driver.

Reply:
This was also why I omitted the check initially but without the check, `sanity test` will fail for one of the test cases that requires error when `pulishVolume` is called on non-existing node.

Reply:
Ah, I see. K8s client QPS is a significant problem right now. I suggest one of the following:

1. Remove the gets, disable the sanity test case and file an issue to fix later.
2. Implement the informer-based caching and wait mechanism we discussed in #176 before implementing the async attach/detach.

Since the client-side QPS is already a known issue that the informer-based caching would address, I favor taking approach 2. I am open to other ideas.

Reply:
Yup, I presumed that with the informer caching, we would be able to minimize the overhead from the api-server GETs. I also would prefer implementing informer caching :)